### ISSUE TITLE
Bluesky hung at 932 Acquisitions

### ISSUE THREAD

**Issue by qzhang234** [aps-8id-dys/ipython-8idiuser#187]
There's space on /home/8-id-i-stage, the helper is running, the detector is ready and waiting for trigger signal.

<img width="1068" alt="Screen Shot 2020-06-27 at 9 15 21 PM" src="https://user-images.githubusercontent.com/48140482/85935911-556fb200-b8bb-11ea-8b34-29d8454ba94b.png">
<img width="1248" alt="Screen Shot 2020-06-27 at 9 14 28 PM" src="https://user-images.githubusercontent.com/48140482/85935898-32dd9900-b8bb-11ea-9f3c-98d8928881b6.png">

**Comment by prjemian** [comment]
Your message was for an error about 9:30 pm so looking at logs near then.  There are a LOT of lines in the log files so it really helps to know where to focus.

First thing I see (in the parent directory) is the SPEC data file: `20200627-174700.dat`.  The last scan starts at 20:36:43 and ends with an abort at 22:38:22:

<details>
<summary>SPEC data file fragment</summary>

```
#S 14694  multi_capil(detector_name=rigaku, acquire_time=1.9e-05, 
acquire_period=1.9e-05, num_images=100000, 
file_name=Q092_QD_Dec_2_att00_0902, submit_xpcs_job=True, 
path=/home/8-id-i/2020-2/comm202006/)
#D Sat Jun 27 20:36:43 2020
#C Sat Jun 27 20:36:43 2020.  plan_type = generator
#C Sat Jun 27 20:36:43 2020.  uid = f7d1d52c-a3c7-4d25-83b5-4be983f04558
#MD uid = f7d1d52c-a3c7-4d25-83b5-4be983f04558
#MD ARun_number = Q092_QD_Dec_2_att00_0902
#MD beamline_id = APS DYS 8-ID-I
#MD file_name = Q092_QD_Dec_2_att00_0902
#MD file_path = /home/8-id-i/2020-2/comm202006/Q092_QD_Dec_2_att00_0902/
#MD login_id = 8idiuser@quartz.xray.aps.anl.gov
#MD pid = 19419
#MD proposal_id = testing
#MD versions = {}
#P0
#N 0
#C no data column labels identified
#C Sat Jun 27 22:38:22 2020.  num_events_baseline = 1
#C Sat Jun 27 22:38:22 2020.  exit_status = abort
```
</details>

That gives me the UID of the start document on the first #MD line: `f7d1d52c-a3c7-4d25-83b5-4be983f04558`

So, I've learned _when_ this happened (shortly after 20:36:43, so I need the log file for this time) and what to look for in the logs: `uid = f7d1d52c`...  That's in file `ipython_logger.log.1` _now_ (as time goes by, this name will change to .2, then .3, and so on until after .9, the file is deleted so that we don't keep too many log files).

And, no log information at the DEBUG level.  Way too bad.  We've decided (#172) to discard something that I believe now would be valuable.

Look for the time or ARun_number (`Q092_QD_Dec_2_att00_0902`) instead:

<details>
<summary>log content for <tt>Q092_QD_Dec_2_att00_0902</tt></summary>

```
|2020-06-27 20:36:43.268|INFO|19419|bluesky-session|bp_CPMV|59|Thread-1| 
- Q092_QD-Dec-2_att00_0902
|2020-06-27 
20:36:43.268|INFO|19419|bluesky-session|xpcs_acquire|48|Thread-1| - 
AD_Acquire starting
|2020-06-27 
20:36:43.268|INFO|19419|bluesky-session|xpcs_acquire|59|Thread-1| - 
file_path = /home/8-id-i/2020-2/comm202006/Q092_QD_Dec_2_att00_0902/
|2020-06-27 
20:36:43.288|INFO|19419|bluesky-session|xpcs_acquire|304|Thread-1| - 
calling full_acquire_procedure()
|2020-06-27 
20:36:43.298|INFO|19419|bluesky-session|xpcs_acquire|146|Thread-1| - 
detNum=46, det_pars={'ccdHardwareColSize': 512, 'ccdHardwareRowSize': 
1024, 'ccdxsense': 1, 'ccdzsense': 1, 'harmonic': 1, 'dpix': 0.076, 
'saturation': 3, 'adupphot': 1.0, 'efficiency': 1.0, 'gain': 1, 
'blemish': 1, 'flatfield': 0, 'distortion': 0, 'parasitic': 0, 'lld': 0, 
'manufacturer': 'RIGAKU500K_NoGap'}
|2020-06-27 
20:36:43.368|INFO|19419|bluesky-session|xpcs_acquire|268|Thread-1| - 
scaler should be autocounting now
|2020-06-27 
20:36:43.485|INFO|19419|bluesky-session|xpcs_acquire|299|Thread-96960| - 
DM workflow kickoff done
|2020-06-27 
20:36:43.486|INFO|19419|bluesky-session|xpcs_acquire|300|Thread-96960| - 
b'id=d1548cd1-9716-4e16-84bc-57098109daad owner=8idiuser status=pending 
startTime=1593308203.46 startTimestamp=2020/06/27 20:36:43 CDT \n'
|2020-06-27 
22:51:51.390|INFO|10765|bluesky-session|session_logs|34|MainThread| - 
############################################################ startup
```

</details>

Compare with messages for previous scan:

<details>
<summary>log content for <tt>Q092_QD_Dec_2_att00_0901</tt></summary>

```
|2020-06-27 20:36:39.396|INFO|19419|bluesky-session|bp_CPMV|59|Thread-1| 
- Q092_QD-Dec-2_att00_0901
|2020-06-27 
20:36:39.396|INFO|19419|bluesky-session|xpcs_acquire|48|Thread-1| - 
AD_Acquire starting
|2020-06-27 
20:36:39.397|INFO|19419|bluesky-session|xpcs_acquire|59|Thread-1| - 
file_path = /home/8-id-i/2020-2/comm202006/Q092_QD_Dec_2_att00_0901/
|2020-06-27 
20:36:39.417|INFO|19419|bluesky-session|xpcs_acquire|304|Thread-1| - 
calling full_acquire_procedure()
|2020-06-27 
20:36:39.426|INFO|19419|bluesky-session|xpcs_acquire|146|Thread-1| - 
detNum=46, det_pars={'ccdHardwareColSize': 512, 'ccdHardwareRowSize': 
1024, 'ccdxsense': 1, 'ccdzsense': 1, 'harmonic': 1, 'dpix': 0.076, 
'saturation': 3, 'adupphot': 1.0, 'efficiency': 1.0, 'gain': 1, 
'blemish': 1, 'flatfield': 0, 'distortion': 0, 'parasitic': 0, 'lld': 0, 
'manufacturer': 'RIGAKU500K_NoGap'}
|2020-06-27 
20:36:39.493|INFO|19419|bluesky-session|xpcs_acquire|268|Thread-1| - 
scaler should be autocounting now
|2020-06-27 
20:36:39.593|INFO|19419|bluesky-session|xpcs_acquire|299|Thread-96873| - 
DM workflow kickoff done
|2020-06-27 
20:36:39.593|INFO|19419|bluesky-session|xpcs_acquire|300|Thread-96873| - 
b'id=79647539-44b7-40a9-8a07-fc9acb85f409 owner=8idiuser status=pending 
startTime=1593308199.57 startTimestamp=2020/06/27 20:36:39 CDT \n'
|2020-06-27 
20:36:43.225|INFO|19419|bluesky-session|xpcs_acquire|294|Thread-96960| - 
DM workflow kickoff starting: analysis:True 
file:/home/8-id-i/2020-2/comm202006/Q092_QD_Dec_2_att00_0901/Q092_QD_Dec_2_att00_0901_0001-100000.hdf
```

</details>

There is no corresponding message for `DM workflow kickoff starting:` in the scan that hung.  That's unusual.  Did we get an HDF metadata file for this?  Expecting to find this file:

    /home/8-id-i/2020-2/comm202006/Q092_QD_Dec_2_att00_0901/Q092_QD_Dec_2_att00_0902_0001-100000.hdf

Look for that file (in a console):

```
snow% ll 
/home/8-id-i/2020-2/comm202006/Q092_QD_Dec_2_att00_0901/Q092_QD_Dec_2_att00_0902_0001-100000.hdf
ls: cannot access 
/home/8-id-i/2020-2/comm202006/Q092_QD_Dec_2_att00_0901/Q092_QD_Dec_2_att00_0902_0001-100000.hdf: 
No such file or directory
snow% ll 
/home/8-id-i/2020-2/comm202006/Q092_QD_Dec_2_att00_0901/Q092_QD_Dec_2_att00_090*.hdf
-rw-rw-r-- 1 8idiuser s8staff 41896 Jun 27 20:36 
/home/8-id-i/2020-2/comm202006/Q092_QD_Dec_2_att00_0901/Q092_QD_Dec_2_att00_0901_0001-100000.hdf
```

Failed to write the metadata file.  Why?

**Comment by prjemian** [comment]
Why did it fail to write the metadata file?  I see no information to inform us.

**Comment by qzhang234** [comment]
I think we knew that hung was from insufficient disk space on /home/8-id-i-stage. So I made some space and I’m running something right now to see if it hangs again

________________________________

**Comment by prjemian** [comment]
Again?  Is this routine?  Should we try to monitor that (perhaps create a bluesky read-only signal and log that in our plan)?

**Comment by prjemian** [comment]
that: the disk space available on `/home/8-id-i-stage`.

Also, we might re-consider (after #172) changing back to `logger.setLevel(logging.DEBUG)` since the log output to the console will be turned off.  What do you think?

**Comment by qzhang234** [comment]
We could, but I implemented some fix, including changing the trigger signal from BNC to web command like in Spec, and added a 0.1 s wait time after Bluesky sees that the Labview is ready before sending the web command, and after the web command is sent. I documented it in #183 .

I also had Nakaye show me where to check whether XSPA has received the trigger signal. This way both the sending of the trigger signal from Bluesky and the ready signal from XSPA will be a status signal instead of a transient pulse. Hopefully that will help us narrow down the culprit. If it still doesn't work we can change to `logging.DEBUG`.
